### PR TITLE
Document normalmap textures

### LINKED ISSUE #9960: Normal textures are undocumented
##### Minetest version
```
5.2.0
```

###### Description
Minetest supports normal textures by using a name like `example_foobar_normal.png` (the “_normal” is the important part).

The support for `_normal` textures is undocumented in `lua_api.txt`. Please fix this.

See also: #9242

@HybridDog

### PR DESCRIPTION
**Pull request by HybridDog**
Document normalmap textures in lua_api.txt.
This should fix #9960.

## To do

This PR is a Ready for Review.

### CODE REVIEW

**Comment by ClobberXD** [comment]
Glad to see some effort being made in this direction. Thanks.

File: doc/lua_api.txt [hunk 1]
```diff
@@ -378,6 +379,38 @@ stripping out the file extension:
...
+to load normalmap textures.
+Those textures have to end with `_normal.png` and start with the same name
+as their corresponding albedo texture.
+For example a normalmap for `foomod_foothing.png` has to be called
+`foomod_foothing_normal.png`.
+The sRGB R, G and B colour values of a normalmap pixel are each linearly
+mapped from `{0, ..., 255}` to `[-1, 1]` and, taken together,
+define the normal vector.
+The alpha channel defines the heightmap for parallax occlusion.
+To be save, the alpha values should always bigger than zero because the colour
```
Comment (appgurueu) [inline doc/lua_api.txt:396]: safe*

File: doc/lua_api.txt [hunk 2]
```diff
@@ -378,6 +379,38 @@ stripping out the file extension:
...
+the height and normal vectors are roughly approximated with
+the brightness of the albedo texture's pixels.
+
+Bumpmapping and parallax occlusion are currently experimental features:
+
+* Bumpmapping in Minetest happens in an obscure way; there are no light sources
+  defined in the shaders except the sunlight direction.
+* Parallax occlusion with relief-mapping mode does not yet work correctly
+  together with Minetest's Fastfaces.
+* The normalmap files must end with `.png`, so jpeg files are not supported.
```
Comment (appgurueu) [inline doc/lua_api.txt:411]: The normalmap files must end with `.png`, so other image files are not supported.

Comment (HybridDog) [reply doc/lua_api.txt]: Thanks for the suggestions.

**Comment by Wuzzy2** [comment]
I think the term “albedo texture” is too obscure. Maybe replace it?

Apart from this, this PR is good.

**Comment by HybridDog** [comment]
> I think the term “albedo texture” is too obscure. Maybe replace it?

I have change it now.

**Comment by Wuzzy2** [comment]
All is good now. :+1:
:smile:

**Comment by SmallJoker** [comment]
Thanks for documenting the normal map textures. In overall the changes look good, but I don't think bugs should be documented, otherwise the workarounds might become a standard practice for modders even after the issues are fixed.

**Comment by HybridDog** [comment]
> […] I don't think bugs should be documented, otherwise the workarounds might become a standard practice for modders even after the issues are fixed.

I have mentioned that bumpmapping is `experimental`, which should imply that the bugs could be fixed in the future and that lighting methods can change. I think it is important to mention that bumpmapping in Minetest does not use the Phong shading (or something similar) because it ignores the light sources.

**Review by sfan5 — APPROVED** [review]

File: doc/lua_api.txt [hunk 3]
```diff
@@ -378,6 +379,39 @@ stripping out the file extension:
...
+define the normal vector.
+The alpha channel defines the heightmap for parallax occlusion.
+To be safe, the alpha values should always be bigger than zero
+because the colour values, which define the normal vector,
+may be undefined for image formats where colour is discarded in fully
+transparent pixels.
+If Minetest cannot find the normalmap for a texture and the
+`generate_normalmaps` setting is enabled,
+the height and normal vectors are roughly approximated with
+the brightness of the texture's pixels.
```
Comment (HybridDog) [inline doc/lua_api.txt:403]: #10313

File: doc/lua_api.txt [hunk 4]
```diff
@@ -378,6 +379,39 @@ stripping out the file extension:
...
+Normalmap Textures
+------------------
+
+If shaders and bumpmapping or parallax occlusion is enabled, Minetest tries
+to load normalmaps.
+Those image files have to end with `_normal.png` and start with the same name
+as their corresponding texture.
+For example a normalmap for `foomod_foothing.png` has to be called
+`foomod_foothing_normal.png`.
+The sRGB R, G and B colour values of a normalmap pixel are each directly
```
Comment (rubenwardy) [inline doc/lua_api.txt:392]:
```suggestion

The sRGB R, G and B colour values of a normalmap pixel are each directly
```

File: doc/lua_api.txt [hunk 5]
```diff
@@ -378,6 +379,39 @@ stripping out the file extension:
...
+mapped from `{0, ..., 255}` to `[-1, 1]` and, taken together,
+define the normal vector.
+The alpha channel defines the heightmap for parallax occlusion.
+To be safe, the alpha values should always be bigger than zero
+because the colour values, which define the normal vector,
+may be undefined for image formats where colour is discarded in fully
+transparent pixels.
+If Minetest cannot find the normalmap for a texture and the
```
Comment (rubenwardy) [inline doc/lua_api.txt:400]:
```suggestion

If Minetest cannot find the normalmap for a texture and the
```

**Review by rubenwardy — APPROVED** [review]

**Pull request merged**